libsolidity/codegen/ir/IRGenerationContext.h

```diff
@@ -150,6 +150,12 @@ class IRGenerationContext
 	bool inlineAssemblySeen() const { return m_inlineAssemblySeen; }
 	void setInlineAssemblySeen() { m_inlineAssemblySeen = true; }
 
+	int64_t functionId(int64_t _astId) {
+		if (m_functionIds.find(_astId) == m_functionIds.end())
+			m_functionIds[_astId] = ++m_currentFunctionId;
+		return m_functionIds.at(_astId);
+	}
```

Comment:
```suggestion
	int64_t functionId(ASTNode const& _function)
	{
		return m_functionIds.try_emplace(_function.id(), m_functionIds.size() + 1).first;
	}
```

Reply:
We should also add a check that the function being added is actually an internal function. Maybe it would also be better to rename this function to `internalFunctionID()`?